### ISSUE TITLE
Distribute count for a pure distribute volume is always 1

### ISSUE THREAD

**Issue by nbalacha** [gluster/glusterd2#847]
Create a pure distribute volume and check the distribute count - it is always 1. 
This causes rebalance to fail for pure distribute volumes.

**Comment by Madhu-1** [comment]
@nbalacha  can you paste the exact steps to reproduce and the result what you got

**Comment by nbalacha** [comment]
Create a pure distribute volume with 2 bricks.

[root@rhgs313-6 brick2]# curl -X GET http://192.168.122.6:24007/v1/volumes/dist
{"id":"1e6129e5-414b-4911-9c7c-4ebfbd8f583f","name":"dist","type":"Distribute","transport":"tcp",**"distribute-count":1**,"replica-count":0,"arbiter-count":0,"options":{},"state":"Started","subvols":[{"name":"dist-distribute-0","type":"Distribute","bricks":[{"id":"6acb33fe-3b0a-4005-8fb4-fb198623186b","path":"/bricks/brick2/dist-1","volume-id":"1e6129e5-414b-4911-9c7c-4ebfbd8f583f","volume-name":"dist","peer-id":"c020a398-0111-4e19-bb24-6a4fdd484196","host":"192.168.122.6","type":"Brick"},{"id":"583f495b-4163-4497-8845-533521addcf0","path":"/bricks/brick2/dist-2","volume-id":"1e6129e5-414b-4911-9c7c-4ebfbd8f583f","volume-name":"dist","peer-id":"c020a398-0111-4e19-bb24-6a4fdd484196","host":"192.168.122.6","type":"Brick"}],"replica-count":1,"arbiter-count":0}],"metadata":{}}

It should be 2.

Note: this was the case some time ago. I have not checked the latest code.

**Comment by Madhu-1** [comment]
Test with new code and let me know if it's not working,

am not able to reproduce  it

```
http://10.70.43.235:24007/v1/volumes
  
{
    "id": "0cd12229-d851-47f4-9e9c-754fcbec5c76",
    "name": "test_volume",
    "type": "Distribute",
    "transport": "tcp",
    "distribute-count": 2,
    "replica-count": 0,
    "arbiter-count": 0,
    "options": {},
    "state": "Created",
    "subvols": [
        {
            "name": "test_volume-distribute-0",
            "type": "Distribute",
            "bricks": [
                {
                    "id": "901c551b-d940-4a8d-b51d-a57799a87dac",
                    "path": "/export/brick11/data",
                    "volume-id": "0cd12229-d851-47f4-9e9c-754fcbec5c76",
                    "volume-name": "test_volume",
                    "peer-id": "4c4b441b-1999-437c-9ba7-9fc3ca5f93d0",
                    "host": "10.70.42.235",
                    "type": "Brick"
                }
            ],
            "replica-count": 1,
            "arbiter-count": 0
        },
        {
            "name": "test_volume-distribute-1",
            "type": "Distribute",
            "bricks": [
                {
                    "id": "ff8e6f44-aff2-46c3-afcf-6c0a2d485119",
                    "path": "/export/brick21/data",
                    "volume-id": "0cd12229-d851-47f4-9e9c-754fcbec5c76",
                    "volume-name": "test_volume",
                    "peer-id": "fef969d9-0c5f-4f2e-9b07-1f65064275ef",
                    "host": "dhcp43-235.lab.eng.blr.redhat.com",
                    "type": "Brick"
                }
            ],
            "replica-count": 1,
            "arbiter-count": 0
        }
    ],
    "metadata": {},
    "snap-list": null
}
```

**Comment by aravindavk** [comment]
I fixed this issue with CLI https://github.com/gluster/glusterd2/pull/797

> This does not match the volfile from GD1. There is only one dht xlator instance for every volume. Having this naming convention can cause other operations to fail (rebalance and so on)

Volfile generation should work fine with the above CLI change. Please confirm with latest master.

**Comment by atinmu** [comment]
@nbalacha - Is this something what you can retest and report back?

**Comment by aravindavk** [comment]
The issue is fixed by the PR #797 as mentioned in the previous comment. Closing this issue. Please reopen if this issue found again.
